frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/obc-test.spec.ts

```diff
@@ -0,0 +1,165 @@
+        expect(arrEl).to.equal(MASK);
+      });
+      expect(secrets.length).to.equal(childCount);
+    });
+  });
+
+  it('Test if secret data can be revealed', () => {
+    obcHandler.revealHiddenValues();
+    cy.byTestID('co-copy-to-clipboard__text').then(($el) => {
+      cy.byTestID('secret-data__dt').should(($h) => {
+        expect($h[0].innerText).to.equal('Endpoint');
+        expect($el[0].innerText).to.include(NS);
+        expect($h[2].innerText).to.equal('Access Key');
+        expect($el[2].innerText).to.match(new RegExp(ACCESS_KEY));
+        expect($h[3].innerText).to.equal('Secret Key');
+        expect($el[3].innerText).to.match(new RegExp(SECRET_KEY));
+      });
+    });
+  });
+
+  it('Test if secret data can be hidden again', () => {
+    obcHandler.hideValues();
+    cy.byTestID('co-copy-to-clipboard__text').as('secrets');
+    cy.get('@secrets').should(($el) => {
+      const childCount = $el.length;
+      const hiddenText = 'Value hidden';
+      const elText = $el.text();
+      const secrets = elText.split(hiddenText).slice(1);
+      secrets.forEach((arrEl) => {
+        expect(arrEl).to.equal(MASK);
+      });
+      expect(secrets.length).to.equal(childCount);
+    });
+  });
+
+  it('Test if labels and annotations are shown correctly', () => {
+    cy.byTestID('label-list').contains(NOOBAA_LABEL);
+    cy.byTestSelector('details-item-value__Annotations').contains(NO_ANNOTATIONS);
+  });
+
+  it('Test if namespace and secret are shown correctly', () => {
+    obcHandler.assertNamespaceExists();
+    cy.get('[title="Secret"]')
+      .siblings()
+      .contains(OBC_NAME);
+  });
+
+  it('Test if status and storage class are shown correctly', () => {
+    cy.byTestID('status-text').contains(BOUND);
+    cy.byLegacyTestID('openshift-storage.noobaa.io').contains(OBC_STORAGE_CLASS_EXACT);
+  });
+
+  it('Test if Object Bucket is created', () => {
+    const obName = `obc-${testName}-${OBC_NAME}`;
+    cy.contains(obName).click();
+    cy.byTestID('co-m-loader').should('not.exist');
+    cy.byLegacyTestID('resource-title').as('title');
+    cy.get('@title')
+      .should('be.visible')
+      .then(($el) => {
+        expect($el.text()).to.equal(obName);
+      });
+    cy.byTestID('resource-status').as('status');
+    cy.get('@status')
+      .should('be.visible')
+      .then(($el) => {
+        expect($el.text()).to.equal(BOUND);
+      });
+  });
+
+  it('Test attachment to a Deployment', () => {
+    cy.clickNavLink(['Storage', 'Object Bucket Claims']);
+    cy.get('tr')
+      .contains(OBC_NAME)
+      .should('be.visible')
+      .byLegacyTestID('kebab-button')
+      .first()
+      .click();
+    cy.contains(ATTACH_TO_DEPLOYMENT).click();
+    cy.byTestID('dropdown-selectbox').click();
+    cy.contains(testDeployment).click();
+    cy.contains('button', 'Attach').click();
+    obcHandler.deploymentReady(testDeployment);
+  });
+
+  after(() => {
+    cy.visit(obcUrl);
```

Comment:
```suggestion
```

Reply:
Removing this fails the `after` block as the page that deletes the OBC is not visited. This is because the second `it` block (not sure about the first one) navigates to a different page during its execution. Let me look a bit further into this and get back to you.

Reply:
Okay so I looked it up and the lines below navigate the respective tests to a different URL, which is why we need to ensure that we are back at the OBC page when we enter the `after` block so as to delete the Bucket Claim.

First test: https://github.com/openshift/console/pull/8061/files#diff-bc97b04fc2ffc9319572ed03433dc859d1ffacac875f13eb19c0df1874650ffaR102

Second test: https://github.com/openshift/console/pull/8061/files#diff-bc97b04fc2ffc9319572ed03433dc859d1ffacac875f13eb19c0df1874650ffaR114